### ISSUE TITLE
no-duplicate-case rule incorrectly flags case statements that use a property reference

### ISSUE THREAD

**Issue by bedney** [eslint/eslint#2052]
As of 0.17.0, the new `no-duplicate-case` rule incorrectly flags the following `case` constructs:

```
var MSGS = {
    hello: 'hi',
    goodbye: 'bye'
};

function lookupMessage(msg) {

    switch (msg) {
        case MSGS.hello:
            return 'hi there';
        case MSGS.goodbye:
            return 'catch ya later';
    }
}

lookupMessage('hi');
```

**Comment by doberkofler** [comment]
This is a duplicate of #2038

**Comment by bedney** [comment]
Oops. Thanks!
